pkgs/build-support/writers/default.nix

```diff
@@ -195,6 +198,23 @@ rec {
     ${pkgs.gixy}/bin/gixy $out
   '';
 
+  # writeNix takes a name a name and content and returns a file which was parsed
+  # with `nix-instantiate --parse --show-trace`
+  #
+  # Example:
+  #   writeNix "example" ''
+  #     {
+  #       hello = "world";
+  #     }
+  #   ''
+  writeNix = name: text: pkgs.runCommand name {
+    inherit text;
+    passAsFile = [ "text" ];
+  } ''
+    ${pkgs.nix}/bin/nix-instantiate --parse --show-trace "$textPath"
+    cp "$textPath" $out
+  '';
```

Comment:
This change seems to have nothing to do with the PR title

Reply:
ah whupps, this WIP and shouldn't be here (^o^;